### PR TITLE
Add Code coverage badge to the repo

### LINKED ISSUE #692: Add code coverage badge to repo
Code coverage is already set up, but something needs to get the data for the badge. 

Some things to look into:
- https://smarie.github.io/python-genbadge/
- https://github.com/marketplace/actions/coverage-py-badge

### PR DESCRIPTION
**Pull request by mihrab34**
to fix #692 

django-unicorn – README md 6_21_2024 4_57_51 PM

### CODE REVIEW

**Comment by mihrab34** [comment]
@adamghill

**Comment by mihrab34** [comment]
I also generated the coverage report in HTML, in case you would like to look at it later.

**Comment by adamghill** [comment]
Thank you! This looks great! I assume this would require manually running `genbadge` and committing the output? Should that be added to docs somewhere? Maybe the new CONTRIBUTING.md file I am going to make?

Did you see any way to automate updating a README.md with a calculated badge on every merge into `main`? I found https://dev.to/thejaredwilcurt/coverage-badge-with-github-actions-finally-59fa (which looks like a real hack) and https://github.com/marketplace/actions/pytest-coverage-comment#auto-updating-badge-on-readme (which maybe isn't as bad)?

**Comment by mihrab34** [comment]
> Thank you! This looks great! I assume this would require manually running `genbadge` and committing the output? Should that be added to docs somewhere? Maybe the new CONTRIBUTING.md file I am going to make?
> 
> Did you see any way to automate updating a README.md with a calculated badge on every merge into `main`? I found https://dev.to/thejaredwilcurt/coverage-badge-with-github-actions-finally-59fa (which looks like a real hack) and https://github.com/marketplace/actions/pytest-coverage-comment#auto-updating-badge-on-readme (which maybe isn't as bad)?

I did read something about running the whole process with GitHub actions. I'd look at the links you shared and add that to the implementation. Thanks!

**Comment by mihrab34** [comment]
@adamghill Instead of using GitHub gist, I added the last step in the workflow as a commit process, as I wanted to keep all the badging processes within the repository.

**Comment by adamghill** [comment]
Makes sense! Yeah the gist process seemed pretty strange. I'll take a look in detail soon, thank you!

**Comment by adamghill** [comment]
It's a bummer there needs to be a commit just for updating the coverage number in the readme, but it looks like https://github.com/MishaKav/pytest-coverage-comment/blob/main/.github/workflows/update-coverage-on-readme.yml does basically the same thing. 🤷 Maybe no good way around it.

This looks great! Thanks for setting it all up. 🚀

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Coverage
+on: [push]
```
Comment (adamghill) [inline .github/workflows/coverage.yml:2]: I think we should only run this on push to `main`. What do you think? https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-filters

**Review by adamghill — CHANGES_REQUESTED** [review]
This looks good, I think we should only run it for pushes to `main`, but other than that I think it can be merged in!

**Comment by mihrab34** [comment]
@adamghill updated to run only on push to `main`

**Review by adamghill — APPROVED** [review]
🚀

**Comment by adamghill** [comment]
@all-contributors add mihrab34 for code

**Pull request merged**